chronos/streams/chunkstream.nim

```diff
@@ -0,0 +1,383 @@
+#
+#    Chronos Asynchronous Chunked-Encoding Stream
+#             (c) Copyright 2019-Present
+#         Status Research & Development GmbH
+#
+#              Licensed under either of
+#  Apache License, version 2.0, (LICENSE-APACHEv2)
+#              MIT license (LICENSE-MIT)
+import ../asyncloop, ../timer
+import asyncstream, ../transports/stream, ../transports/common
+export asyncstream, stream, timer, common
+
+const
+  ChunkBufferSize = 4096
+  ChunkHeaderSize = 8
+  ChunkDefaultTimeout = 10.seconds
+  CRLF = @[byte(0x0D), byte(0x0A)]
+
+type
+  ChunkedStreamReader* = ref object of AsyncStreamReader
+    timeout: Duration
+
+  ChunkedStreamWriter* = ref object of AsyncStreamWriter
+
+  ChunkedStreamError* = object of Exception
+  ChunkedStreamTimeoutError* = object of ChunkedStreamError
+  ChunkedStreamProtocolError* = object of ChunkedStreamError
+
+proc newTimeoutError(): ref Exception {.inline.} =
+  newException(ChunkedStreamTimeoutError, "Timeout exceeded!")
+
+proc newProtocolError(): ref Exception {.inline.} =
+  newException(ChunkedStreamProtocolError, "Protocol error!")
+
+proc oneOf*[A, B, C](fut1: Future[A], fut2: Future[B],
```

Comment:
Shouldn't this be moved in a more general helpers module?

Reply:
I'm not sure if it will be usable in other modules, but if it will be it will be moved of course.